### ISSUE TITLE
extracted libraries (DLL etc) not deleted properly on exit

### ISSUE THREAD

**Issue by FyiurAmron** [xerial/sqlite-jdbc#87]
see https://bitbucket.org/xerial/sqlite-jdbc/issues/202/jdbc-driver-extracts-a-new-dll-library-on#comment-None - firstly, http://bugs.java.com/view_bug.do?bug_id=6357433 - but since we have `                if(writer != null) {
                    writer.close();
                }` and it ain't closed, maybe it's due to the fact that the DLL has been loaded and is not unloaded? ( https://web.archive.org/web/20140704120535/http://www.codethesis.com/blog/unload-java-jni-dll as the possible solution in that case )

Anyway, to allow a manual workaround: maybe keep "extractedLibFile" somewhere, so that the API user can actually *know* what is the lib (DLL) that has been extracted, to allow later cleanup for it (external script or whatever).

**Comment by promille** [comment]
I had the same Problem with tons of sqlite.dll's in my tmp dir.

I tried to delete the dll's via an Runtime shutdownHook but the dll's can not be delete.
Yes I think its because the dll is attached to the current running process.

Now I removed the uuid from the extrected dll name and test if the dll is already there -> use it / otherwise copy and use it.

So there will be one dll for each version only without any deletion.
